### PR TITLE
Tsylia / Fixed the favorite workshops doesn't appear in Favorites for current …

### LINKED ISSUE #1160: [Add to favorites] The workshop does not appear in the favorites tab after the user clicks the add to favorites button.
**Environment:** Windows 10, 64-bit, Google Chrome 114.0.5735.199,
Windows 10, 64-bit, Microsoft Edge 114.0.1823.67.

**Reproducible:** always.

**Preconditions**

1. Go to https://oos.dmytrominochkin.cloud/#/
2. Sign in as a parent.

**Steps to reproduce**
1. Find a workshop that you need to add to your favorites.
2. Click on the "Add to favorites" button.
3. Go to the "Favorite" page and check if the workshop has appeared.

**Actual result**
The workshop does not appear on the "Favorites" page.

**Expected result**
After adding a workshop to favorites, the workshop is displayed on the "Favorites" page.

image
image
image

**User story and test case links**
[Add to favorites #208] https://github.com/ita-social-projects/OoS-Backend/issues/208

**Labels to be added**
"Bug", Priority ("pri: medium"), Severity ("severity: medium"), Type ("Functional").

### PR DESCRIPTION
**Pull request by andriitsylia**
…user.

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public async Task<IActionResult> GetById(long id)
     [HttpGet]
     public async Task<IActionResult> GetAllByUser()
     {
-        string userId = User.FindFirst("sub")?.Value;
+        string userId = User.FindFirst("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier")?.Value;
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:84]: Better to use ClaimTypes.NameIdentifier

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs [hunk 2]
```diff
@@ -103,7 +103,7 @@ public async Task<IActionResult> GetAllByUser()
     [HttpGet("workshops")]
     public async Task<IActionResult> GetFavoriteWorkshopsByUser([FromQuery] OffsetFilter offsetFilter)
     {
-        string userId = User.FindFirst("sub")?.Value;
+        string userId = User.FindFirst("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier")?.Value;
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:108]: Same

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:84]: The another way - IdentityResourceClaimsTypes.Sub

Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:84]: It doesn't work.

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs (hunk 2 above)
Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:108]: It doesn't work.

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:84]: I don't believe you 😭

Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:84]: Yes, it works.  :)

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs (hunk 2 above)
Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/FavoriteController.cs:108]: Yes, it works.  :)

**Review by alrom — APPROVED** [review]
Now I like it 😁

**Review by DmyMi — APPROVED** [review]

**Pull request merged**